framework/include/vectorpostprocessors/LineMaterialSamplerBase.h

```diff
@@ -179,6 +184,7 @@ LineMaterialSamplerBase<T>::execute()
       addSample(_q_point[qp], qp_proj_dist_along_line, values);
     }
   }
+  _fe_problem.clearActiveMaterialProperties(_tid);
```

Comment:
You are both correct.  The computational cost issue has been refuted, but I'd like to see if we can think of a better way for object not within these automatically handled thread loop cases to at least know they need to init material properties.  Maybe something like a warning if they try to compute material properties but zero/none are actually computed?